pyqir-jit/README.md

```diff
@@ -1,22 +1,75 @@
 # pyqir-jit
 
-## Building and Testing
+## Examples
+
+There are JIT examples in the repository:
+- [Bernstein-Vazirani](../examples/jit/bernstein_vazirani.py) ([Bernstein-Vazirani QIR](../examples/jit/bernstein_vazirani.ll))
+
+Let's walk though the example. Suppose we have the generated IR (`bernstein_vazirani.ll`). We can evaluate it with the `NonadaptiveJit`, and `GateLogger` to print out a simple log of the quantum application.
+
+```python
+from pyqir_jit import NonadaptiveJit, GateLogger
+
+from pathlib import Path
+import os
+
+path = Path(__file__).parent
+file = os.path.join(path, "bernstein_vazirani.ll")
 
-To build this package, first install `maturin`:
+jit = NonadaptiveJit()
+logger = GateLogger()
 
-```shell
-pip install maturin
+print("# output from NonadaptiveJit returning the uninitialized output")
+jit.eval(file, logger)
+
+print("# output from GateLogger")
+logger.print()
 ```
 
-To build and test use `maturin develop`:
+Would generate the output:
 
-```shell
-pip install -r requirements-dev.txt
-maturin develop && pytest
 ```
+# output from NonadaptiveJit returning the uninitialized output
+[[Zero, Zero, Zero, Zero, Zero, Zero, Zero, Zero]]
+# output from GateLogger
+qubits[9]
+out[9]
+x qubit[8]
+h qubit[0]
+h qubit[1]
+h qubit[2]
+h qubit[3]
+h qubit[4]
+h qubit[5]
+h qubit[6]
+h qubit[7]
+h qubit[8]
+cx qubit[2], qubit[8]
+cx qubit[3], qubit[8]
+h qubit[0]
+h qubit[1]
+h qubit[2]
+h qubit[3]
+h qubit[4]
+h qubit[5]
+h qubit[6]
+h qubit[7]
+measure qubits[0] -> out[0]
+measure qubits[1] -> out[1]
+measure qubits[2] -> out[2]
+measure qubits[3] -> out[3]
+measure qubits[4] -> out[4]
+measure qubits[5] -> out[5]
+measure qubits[6] -> out[6]
+measure qubits[7] -> out[7]
+measure qubits[8] -> out[8]
+```
+
+## Building and Testing
+
+See [Building](../docs/building.md)
 
-Alternatively, install tox and run the tests inside an isolated environment:
+## Limitations
 
-```shell
-tox -e py
-```
\ No newline at end of file
+- QIR entrypoint for JIT must be named `QuantumApplication__Run`
```

Comment:
What happens if the entry point takes arguments?
What happens if there is a branching based on a measurement result?

Reply:
I added a line that states that entry point arguments are not yet supported.